cluster-autoscaler/dynamic/config_fetcher.go

```
@@ -0,0 +1,44 @@
+package dynamic
+
+import (
+	"fmt"
+	kube_client "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
+)
+
+// ConfigFetcher fetches the up-to-date dynamic configuration from the apiserver
+type ConfigFetcher struct {
+	configMapName string
+	namespace     string
+	kubeClient    kube_client.Interface
+	lastConfig    Config
+}
+
+// Builds a config fetcher from the parameters and dependencies
+func NewConfigFetcher(configMapName string, namespace string, kubeClient kube_client.Interface) *ConfigFetcher {
+	return &ConfigFetcher{
+		configMapName: configMapName,
+		namespace:     namespace,
+		kubeClient:    kubeClient,
+		lastConfig:    NewDefaultConfig(),
+	}
+}
+
+// Returns the config if it has changed since the last sync. Returns nil if it has not changed.
+func (c *ConfigFetcher) FetchConfigIfUpdated() (*Config, error) {
+	cm, err := c.kubeClient.Core().ConfigMaps(c.namespace).Get(c.configMapName)
```

I would rather raise a loud error. There are couple reasons why the config map is not there, like
* User misspelled the name/namespace
* User has not added it yet.
But in none of the cases i can think of adding an empty map actually helps the user.